### ISSUE TITLE
tests fail on windows

### ISSUE THREAD

**Issue by b-strauss** [angular/tsickle#432]
I just ran gulp test on a windows 10 machine. 73 tests failed. This is probably a path problem.

Examples:
```
69) golden tests variables:
   Error: unexpected file read of test_files/variables/variables.ts not in test_files\variables\variables.ts
    at Object.host.getSourceFile (D:\workspace\b-strauss\test\test_support.ts:64:11)
    at findSourceFile (node_modules\typescript\lib\typescript.js:63447:29)
    at processSourceFile (node_modules\typescript\lib\typescript.js:63378:27)
    at processRootFile (node_modules\typescript\lib\typescript.js:63265:13)
    at node_modules\typescript\lib\typescript.js:62614:60
    at Object.forEach (node_modules\typescript\lib\typescript.js:1293:30)
    at Object.createProgram (node_modules\typescript\lib\typescript.js:62614:16)
    at Object.createProgram (D:\workspace\b-strauss\test\test_support.ts:67:13)
    at Context.<anonymous> (D:\workspace\b-strauss\test\tsickle_test.ts:94:33)

1) source maps handles input source maps:
   Error: unexpected file read of node_modules\typescript\lib\lib.es6.d.ts not in input.ts
    at Object.host.getSourceFile (D:\workspace\b-strauss\test\test_support.ts:64:11)
    at findSourceFile (node_modules\typescript\lib\typescript.js:63447:29)
    at processSourceFile (node_modules\typescript\lib\typescript.js:63378:27)
    at processRootFile (node_modules\typescript\lib\typescript.js:63265:13)
    at Object.createProgram (node_modules\typescript\lib\typescript.js:62634:21)
    at Object.createProgram (D:\workspace\b-strauss\test\test_support.ts:67:13)
    at decoratorDownlevelAndAddInlineSourceMaps (D:\workspace\b-strauss\test\e2e_source_map_test.ts:293:29)
    at Context.<anonymous> (D:\workspace\b-strauss\test\e2e_source_map_test.ts:235:30)
```

It tries to read files from a path outside the project folder.

I tried normalizing paths in the `host.directoryExists` function but failed.

**Comment by mprobst** [comment]
I think the problem is likely that we don't canonicalize paths in here:
https://github.com/angular/tsickle/blob/master/test/test_support.ts#L42

A potential fix would be to use `host.getCanonicalFileName(...)` to canonicalize all keys (i.e. paths) in the map that's being passed in creating a copy of the original map but with `/` forward slashes in its keys.

Care to send a fix if that works? I don't have a Windows computer around to test...

**Comment by evmar** [comment]
In general, we don't develop on Windows so it's unsurprising this doesn't work.  I also don't have a Windows computer so it'll probably fall on users like you to fix this.  :)

**Comment by b-strauss** [comment]
The paths inside the `sources` Map are already canonicalized. I still think the problem is inside the `createProgram` function.

What is `host.directoryExists` supposed to do?
https://github.com/angular/tsickle/blob/master/test/test_support.ts#L46

Should it return false for every path that has `node_modules/@types` inside it?

**Comment by evmar** [comment]
That code is trying to prevent our test suite from being affected by the `node_modules/@types` directory in the tsickle repo.  I think we could better express this requirement by adding an attribute to the `CompilerOptions` used, as is done here
https://github.com/google/kythe/blob/a75326ed3e6920ee819974bad4a6b93d4047eb8d/kythe/typescript/testdata/tsconfig.json#L5

and then deleting this code

**Comment by b-strauss** [comment]
@evmar explicitly setting `types: []` worked, but now it tries to load `tsickle\node_modules\typescript\lib\lib.es6.d.ts` as test file. I tried setting `noLib: true`, but that of course crashed the compilation process. :(

**Comment by b-strauss** [comment]
I don't understand why this is not also a problem on other platforms.

**Comment by b-strauss** [comment]
Closed by #435 

Note that tests only load/run on windows for now. There are still tests that fail.
